### PR TITLE
Starlette: capture custom request response headers in span attributes

### PR DESCRIPTION
**Pull request by sanketmehta28**
# Description

This PR contains py tests for Starlette framework. Actual code has been merged in this PR: https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1004
It also contains the doc update for Startlette framework

Fixes # (917)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Added below unit tests for request and response headers for both http and websocket applications.

- [x] Test A: check for request header which is set in OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST env variable, is present in the span as attribute
- [x] Test B: check for request header which is NOT set in OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST env variable, is NOT present in the span as attribute
- [x] Test C: check for response header which is set in OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_RESPONSE env variable, is present in the span as attribute
- [x] Test D: check for response header which is NOT set in OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_RESPONSE env variable, is NOT present in the span as 
attribute

- Also added test for NoOpTracerProvider as well

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

**Comment by sanketmehta28** [comment]
@open-telemetry/python-approvers:  Please do review this PR and if it looks fine, please do merge it

**Comment by sanketmehta28** [comment]
@open-telemetry/python-approvers: All conflicts are resolved. Please do review this

File: instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py [hunk 1]
```diff
@@ -91,57 +91,6 @@ def client_response_hook(span: Span, message: dict):
...
-
-Request headers
-***************
-To capture predefined HTTP request headers as span attributes, set the environment variable ``OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST``
-to a comma-separated list of HTTP header names.
-
-For example,
-
-::
-
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py:106]: Why are you deleting this?

File: instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py [hunk 2]
```diff
@@ -86,57 +86,6 @@ def response_hook(span, req, resp):
...
-    export OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_RESPONSE="content-type,custom_response_header"
-
-will extract ``content-type`` and ``custom_response_header`` from response headers and add them as span attributes.
-
-It is recommended that you should give the correct names of the headers to be captured in the environment variable.
-Response header names captured in falcon are case insensitive. So, giving header name as ``CUStomHeader`` in environment variable will be able capture header with name ``customheader``.
-
-The name of the added span attribute will follow the format ``http.response.header.<header_name>`` where ``<header_name>`` being the normalized HTTP header name (lowercase, with - characters replaced by _ ).
-The value of the attribute will be single item list containing all the header values.
-
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py:133]: Same

File: instrumentation/opentelemetry-instrumentation-tornado/src/opentelemetry/instrumentation/tornado/__init__.py [hunk 3]
```diff
@@ -97,57 +97,6 @@ def client_resposne_hook(span, future):
...
-For example,
-
-::
-
-    export OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST="content-type,custom_request_header"
-
-will extract ``content-type`` and ``custom_request_header`` from request headers and add them as span attributes.
-
-It is recommended that you should give the correct names of the headers to be captured in the environment variable.
-Request header names in tornado are case insensitive. So, giving header name as ``CUStomHeader`` in environment variable will be able capture header with name ``customheader``.
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-tornado/src/opentelemetry/instrumentation/tornado/__init__.py:118]: Same thing everywhere

**Review by srikanthccv — CHANGES_REQUESTED** [review]

**Review by owais — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py [hunk 4]
```diff
@@ -142,6 +142,9 @@ def client_response_hook(span: Span, message: dict):
 Note:
     Environment variable names to caputre http headers are still experimental, and thus are subject to change.
 
+API
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py:145]: Duplicate?

File: instrumentation/opentelemetry-instrumentation-fastapi/tests/test_fastapi_instrumentation.py [hunk 5]
```diff
@@ -691,4 +691,4 @@ def test_custom_header_not_present_in_non_recording_span(self):
         )
         self.assertEqual(200, resp.status_code)
         span_list = self.memory_exporter.get_finished_spans()
-        self.assertEqual(len(span_list), 0)
+        self.assertEqual(len(span_list), 0)
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-fastapi/tests/test_fastapi_instrumentation.py:694]: Add a blank line here.

**Comment by sanketmehta28** [comment]
@lzchen: all your comments are addressed. Please do review and merge this PR

File: instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py [hunk 6]
```diff
@@ -244,3 +255,334 @@ def test_mark_span_internal_in_presence_of_another_span(self):
...
+        resp = self._client.get("/foobar")
+        self.assertEqual(200, resp.status_code)
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 3)
+
+        server_span = [
+            span for span in span_list if span.kind == SpanKind.SERVER
+        ][0]
+
+        for key, _ in not_expected.items():
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py:387]:
```suggestion
        for key in not_expected:
```

**Review by srikanthccv — APPROVED** [review]

**Review by lzchen — APPROVED** [review]

**Pull request merged**